src/sentry/south_migrations/0409_backfill_release_project_environment_last_deploy_id.py

```diff
@@ -0,0 +1,1165 @@
+# -*- coding: utf-8 -*-
+from south.utils import datetime_utils as datetime
+from south.db import db
+from south.v2 import DataMigration
+from django.db import models
+from sentry.utils.query import RangeQuerySetWrapperWithProgressBar
+
+
+class Migration(DataMigration):
+
+    # Flag to indicate if this migration is too risky
+    # to run online and needs to be coordinated for offline
+    is_dangerous = True
+
+    def forwards(self, orm):
+        db.commit_transaction()
+        try:
+            self._forwards(orm)
+        except Exception:
+            db.start_transaction()
+            raise
+        db.start_transaction()
+
+    def _forwards(self, orm):
+        "Write your forwards methods here."
+        # Note: Don't use "from appname.models import ModelName".
+        # Use orm.ModelName to refer to models in this application,
+        # and orm['appname.ModelName'] for models in other applications.
+        release_project_environments = orm.ReleaseProjectEnvironment.objects.all()
```

Comment:
I'd also add the `last_deploy_id__isnull=True` to this instead of just all().

This is becuase we'll be starting to write data before this migration runs. And we don't need to operate over data that has already been set, nor do we want to.